src/decode.h

```diff
@@ -1508,7 +1515,7 @@ static inline bool DecodeNetworkLayer(ThreadVars *tv, DecodeThreadVars *dtv,
             break;
         default:
             SCLogDebug("unknown ether type: %" PRIx16 "", proto);
-            StatsIncr(tv, dtv->counter_ethertype_unknown);
+            ENGINE_SET_UNKNOWN_EVENT(p, ETHERNET_UNKNOWN_ETHERTYPE);
```

Comment:
why not set the event here and increment the counter. Seems to make the flag obsolete

Reply:
Is there a value in having 2 counters for an unknown ethernet type?

The decoder counts occurrences of unknown ethertypes; the same occurrence is counted twice with the event.

Eg.,
```
decoder.unknown_ethertype                                    | Total                     | 281
decoder.event.ethernet.unknown_ethertype                     | Total                     | 281
```

Reply:
how do we handle this for other cases?

Reply:
The other decoders count occurrences, e.g., ethernet, vntag, etc with events for discrete conditions (e.g., header too small).

The unknown case is unique -- we added a counter (Dec 2022) for unknown ether types. However, the counter seems unnecessary with the event tracking the same value.

Can we keep the duality for 8 (maybe?) and remove the counter in 9 leaving behind only the event?

Reply:
This approach makes sense to me.

Reply:
I've created an issue to track removing the non-event based counter for post 8.0: https://redmine.openinfosecfoundation.org/issues/7528